### PR TITLE
Change behavior of loading new posts in the reader

### LINKED ISSUE #2133: Reader should remember where the user left off
In the iOS app, when the user returns to the reader it loads new content above where you left off, so it stays scrolled to the post you were last looking at. Consider mimicking this in the Android app, perhaps with some indication that newer posts exist.

### PR DESCRIPTION
**Pull request by nbradbury**
Fix #2133 - rewrote how new posts are loaded and displayed in the reader. When new posts are loaded, the current position is retained and a "New posts" bar appears at the top of the screen which animates out after a few seconds. Tapping this bar scrolls the new posts into view.

Making this work required avoiding the use of `notifyDataChanged()` in the adapter, since that would lose the current position (and was very inefficient). The [new code](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/2133-reader-remember-position/WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java#L625) relies on `notifyItemInserted()` and `notifyItemChanged()` instead.

### CODE REVIEW

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -665,7 +665,7 @@
     <string name="reader_hint_reblog_comment">Add your thoughts (optional)</string>
 
     <!-- TextView labels -->
-    <string name="reader_label_new_posts">Show new posts</string>
+    <string name="reader_label_new_posts">&#8593; New posts</string>
```
Comment (roundhill) [inline WordPress/src/main/res/values/strings.xml]: The up arrow looks a little weird (thin and too high imo), could we use a `drawableLeft` with a proper icon?

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java [hunk 2]
```diff
@@ -175,6 +177,8 @@ public void setArguments(Bundle args) {
             if (getPostListType() == ReaderPostListType.TAG_PREVIEW && hasCurrentTag()) {
                 mTagPreviewHistory.push(getCurrentTagName());
             }
+
+            // TODO: save KEY_RESTORE_POSITION
```
Comment (roundhill) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java]: Taking care of this later?

**Comment by nbradbury** [comment]
@roundhill Ready for re-review! I redesigned the "new posts" bar to use a drawable as you (correctly) suggested. That "todo" was a misplaced reminder to me to save the current position in onSaveInstanceState which I already added (so I removed the todo).

**Comment by nbradbury** [comment]
@roundhill Good call on the material up arrow icons. I just switched over to them.

**Comment by roundhill** [comment]
There's been a few times where I've opened the reader or switched tags, when the list was at the top. When I did a PTR, the prompt showed that there were new posts, but tapping it didn't show any new posts. I think the prompt was incorrectly shown.

**Comment by nbradbury** [comment]
I think I found the problem: the "new posts" bar would appear if there were *changed* posts even when there weren't any new ones. Fixed in 4a55292.

**Comment by roundhill** [comment]
Grand, let's :shipit:

**Pull request merged**